### PR TITLE
Add `http_request_performed` diagnostics event

### PR DESCRIPTION
**Pull request by tonidero**
### Description
This adds a new diagnostics event `http_request_performed` with some useful information we would like to track about the http requests performed to our servers.

### CODE REVIEW

File: Sources/Networking/HTTPClient/HTTPClient.swift [hunk 1]
```diff
@@ -518,6 +540,40 @@ private extension HTTPClient {
...
+                                                                               resultOrigin: response.origin,
+                                                                               verificationResult: verificationResult)
+                        // swiftlint:disable:next empty_enum_arguments
+                        case .failure(_):
+                            await diagnosticsTracker.trackHttpRequestPerformed(endpointName: requestPathName,
+                                                                               responseTime: responseTime,
+                                                                               wasSuccessful: false,
+                                                                               responseCode: -1,
+                                                                               resultOrigin: nil,
+                                                                               verificationResult: .notRequested)
```
Comment (jamesrb1) [inline Sources/Networking/HTTPClient/HTTPClient.swift:569]: Again unsure on conventions here but `.notRequested` seems possibly incorrect?

**Review by jamesrb1 — APPROVED** [review]
Looking good!!

File: Sources/Networking/HTTPClient/HTTPClient.swift [hunk 2]
```diff
@@ -518,6 +540,40 @@ private extension HTTPClient {
...
+                                                                               wasSuccessful: true,
+                                                                               responseCode: httpStatusCode,
+                                                                               resultOrigin: response.origin,
+                                                                               verificationResult: verificationResult)
+                        // swiftlint:disable:next empty_enum_arguments
+                        case .failure(_):
+                            await diagnosticsTracker.trackHttpRequestPerformed(endpointName: requestPathName,
+                                                                               responseTime: responseTime,
+                                                                               wasSuccessful: false,
+                                                                               responseCode: -1,
```
[the remark this answers is not in the document]
Comment (tonidero) [reply Sources/Networking/HTTPClient/HTTPClient.swift:567]: `-1` is a convention for network errors, where we don't have a status code. For backend errors, I believe we should still be receiving a success here, but with an error inside, but I will double-check that 👍

File: Sources/Networking/HTTPClient/HTTPClient.swift (hunk 1 above)
Comment (tonidero) [reply Sources/Networking/HTTPClient/HTTPClient.swift:569]: Yeah, the idea here is that upon a network error, the verification doesn't even happen, so we default to `.notRequested`. We could default to something else to differentiate, but we already have the `statusCode` property to be able to differentiate those and filter in grafana as needed. We also avoid adding new cardinality to the grafana counters. Lmk if you feel we should have something else though!

File: Sources/Networking/HTTPClient/HTTPClient.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (tonidero) [reply Sources/Networking/HTTPClient/HTTPClient.swift:567]: Actually, I think you might be right, fixing. Good catch!

**Pull request merged**